lib/beaker/dsl/roles.rb

```diff
@@ -89,6 +89,22 @@ def default
         find_host_with_role :default
       end
 
+      # Determine if host is only an agent, does not have roles 'master',
+      # 'dashboard' or 'database'.
+      #
+      # @return [Boolean]      True if agent-only, false otherwise
+      #
+      # @example Basic usage
+      #     if agent_only(host)
+      #       puts "this host is only an agent!"
+      #     end
+      #
+      def agent_only(host)
+        not_agent = ['dashboard', 'database', 'master', 'console']
+        roles = host['roles'].select { |v| not_agent.include?(v) }
+        roles.length == 0
```

Comment:
@anodelman I'm sure the way you're using this method is fine, but it's actually confusing to a user of the method.

For example, if a host has the roles `['agent', 'frictionless']` (as all agents that should be installed with frictionless should be unless I'm missing something), it will report as agent only.  This would probably be fine for your use case, but as soon as I saw this method, I thought, "oh hey, I should be able to use this for the [pe_util's install method](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/install_utils/pe_utils.rb#L508-L519)!"  Unfortunately it will report that we're agent only in that case, which we aren't.

I would suggest that we either say it doesn't have "higher" roles (hard to define), or we actually make this agent only as in the code example that I referenced.